### PR TITLE
Blog for running gpu based functions on Fission

### PR DESCRIPTION
**Pull request by soharab-ic**

### CODE REVIEW

**Comment by techmaharaj** [comment]
General:
- We can improve the intro -  add some more meat around serverless functions, talk about rise in AI workloads, how serverless can help, here's how we can use  Fission to deploy GPU functions.
- [Important] We should add a section on Why run GPU based functions on Fission, maybe 2-4 points with reasons, that'll be really helpful.
- Let us also mention at the beginning before pre-requisites as to what we are doing in this blog post. Currently it's not mentioned anywhere.
- We can find an opportunity to link to our GPU blog post on InfraCloud.

Specific:
- Let's be clear with pre-requisites, a system with an Nvidia card, Kubernetes cluster, fission installed (link to fission installation doc), Nvidia related stuff installed (link to respective pages) - all these should be bullet points under the sub-heading "Pre-requisites."
- Steps under the "Steps" section are also not clear. 
- Can improve the sentiment example? I mean currently we're hardcoding the statement and the output in the last step just says Positive, and I was where did I give the statement? So two things can be done here,either provide a way to pass the statement as an argument OR print the statement in the output.

File: content/en/blog/running-gpu-based-functions-on-fission.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+  
+  ```bash
+  fission env create --name python --image ghcr.io/soharab-ic/python-pytorch-env --builder ghcr.io/soharab-ic/python-pytorch-builder --poolsize 1
+  ```
+
+- The `fission env create` command will create two deployments. One deployment named `poolmgr-python-default-*` for environment and another for builder named `python-*`.
+- Patch the environment deployment to add GPU resources to `python` environment container and set `nodeSelector` to schedule pods on a GPU node using `kubectl patch` command.
+
+  ```bash
+  kubectl patch deployment poolmgr-python-default-5560759 -p '{"spec": {"template": {"spec":{"containers":[{"name":"python","resources": {"limits": {"nvidia.com/gpu": "1"}, "requests": {"nvidia.com/gpu": "1"}}}]}}}}'
```
Comment (sanketsudake) [inline content/en/blog/running-gpu-based-functions-on-fission.md:177]: Please consider patching environment spec. Not deployment.
User should not change fissions internal object.

Comment (soharab-ic) [reply content/en/blog/running-gpu-based-functions-on-fission.md:177]: @sanketsudake Patching the resources is working via environment spec but nodeSelector patch is not working.
NodeSelector field is ignored when creating deployment.
```
apiVersion: fission.io/v1
kind: Environment
metadata:
  creationTimestamp: null
  name: pytorch
spec:
  builder:
    command: build
    container:
      name: ""
      resources: {}
    image: ghcr.io/fission/python-builder
  imagepullsecret: ""
  keeparchive: false
  poolsize: 1
  resources:
    requests:
      nvidia.com/gpu: "1"
    limits:
      nvidia.com/gpu: "1"
  runtime:
    container:
      name: ""
      resources: {}
    image: ghcr.io/fission/python-env
    podSpec:
    nodeSelector:
      kubernetes.io/hostname: "gpu-node03"
  version: 3
```

Comment (sanketsudake) [reply content/en/blog/running-gpu-based-functions-on-fission.md:177]: So typically you need to have container name same environment. As we have two containers in pod 
- runtime and fetcher 

containers is required field https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#podspec-v1-core

https://github.com/fission/fission/blob/93869d3bc8b6097143e5488366dae178488b4474/test/tests/test_specs/test_spec_merge/specs/env-nodend.yaml#L25

Probably we can skip node selector part in blog post. only keep resources requests.

Comment (sanketsudake) [reply content/en/blog/running-gpu-based-functions-on-fission.md:177]: Checkout 
- https://github.com/fission/fission/pull/2044
- https://github.com/fission/fission/issues/2043

Comment (soharab-ic) [reply content/en/blog/running-gpu-based-functions-on-fission.md:177]: Skipping the node selector part.

Comment (sanketsudake) [reply content/en/blog/running-gpu-based-functions-on-fission.md:177]: Yes please update blog accordingly

**Pull request merged**